test/e2e/service_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// TestClusterIPHostAccess tests traffic from host to Cluster IP Service.
+func TestClusterIPHostAccess(t *testing.T) {
+	// TODO: Support for dual-stack and IPv6-only clusters
+	skipIfIPv6Cluster(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	svcName := "nginx"
+	node := nodeName(0)
+	svc, cleanup := data.createClusterIPServiceAllInOne(t, svcName, node)
+	defer cleanup()
+	time.Sleep(3 * time.Second) // Wait a while for the rules to be installed by kube-proxy/antrea-proxy.
+	t.Logf("%s Service is ready", svcName)
+
+	var linNode, winNode string
+	linNode = node
+	if len(clusterInfo.windowsNodes) != 0 {
+		idx := clusterInfo.windowsNodes[0]
+		winNode = clusterInfo.nodes[idx].name
+	}
+
+	curlSvc := func(node string) {
+		cmd := fmt.Sprintf("curl %s:80", svc.Spec.ClusterIP)
+		rc, stdout, stderr, err := RunCommandOnNode(node, cmd)
+		if rc != 0 || err != nil {
+			t.Errorf("Error when running command '%s' on Node '%s', rc: %d, stdout: %s, stderr: %s, error: %v",
+				cmd, node, rc, stdout, stderr, err)
+		} else {
+			t.Logf("curl from Node '%s' succeeded\n", node)
+		}
+	}
+	t.Logf("Try to curl Cluster IP Service from a Linux host")
+	curlSvc(linNode)
+	if winNode != "" {
+		t.Logf("Try to curl Cluster IP Service from a Windows host")
+		curlSvc(winNode)
+	}
+}
+
+func (data *TestData) createClusterIPServiceAllInOne(t *testing.T, name string, node string) (*corev1.Service, func()) {
```

Comment:
why the name `createClusterIPServiceAllInOne`? Is it because it creates both the backend Pods and the Service? Maybe `createClusterIPServiceAndBackendPods` would be a better name?

Reply:
Yes, both pod and service.
I have updated code to use `createClusterIPServiceAndBackendPods`